### DISCUSSION TITLE
Ability to source values.yaml from a webservice

### DISCUSSION THREAD

**Discussion by dgr237** [argoproj/argo-cd#18479] [General]
I am looking at using ArgoCD and I would like to use generic helm charts. The values.yaml need to be customized based on the Environment that the chart is being installed in e.g. specific configuration based on the AWS Account that the chart is being installed in.

I was thinking of running a webservice in the account which could query the AWS Apis and return the required values.yaml. 

Is it possible using the multiple sources functionality in the Application to be able to invoke a webservice exposed by a service running in the cluster to obtain the values.yaml. I can only find reference in the documentation to sources being configured as either OCI images, Helm repositories or Git repos.

If this is possible could you please provide an example configuration.

**Comment by agaudreault** [comment]
In general, this is an anti-gitops pattern to inject "runtime" configurations in the hydrated/rendered manifest that is deployed in the cluster. I would recommend you to implement this logic as a Mutation Webhook in your clusters.
